Assets/MRTK/SDK/Experimental/Inspectors/ScrollingObjectCollection/ScrollingObjectCollectionInspector.cs

```diff
@@ -311,43 +319,147 @@ private void OnSceneGUI()
+            Vector3 planeLocalPosition = Vector3.up * releaseThresholdTopBottom.floatValue * -1.0f;
+            Vector3 widthHalfExtent = scrollView.transform.right * scrollView.ClippingObject.transform.lossyScale.x / 2;
+            Vector3 heightHalfExtent = scrollView.transform.forward * scrollView.ClippingObject.transform.lossyScale.z / 2;
+
+            Quaternion handleRotation = Quaternion.LookRotation(scrollView.transform.up);
+
+            DrawPlaneAndHandle(planeLocalPosition, widthHalfExtent, heightHalfExtent, handleRotation, releasePlaneColor, BottomReleasePlaneDescription);
+        }
+
+        /// <summary>
+        /// Draws the scroll interaction debug planes.
+        /// </summary>
+        private void DrawPlaneAndHandle(Vector3 planeLocalPosition, Vector3 widthHalfExtent, Vector3 hightHalfExtent, Quaternion handleRotation, Color color, string labelText)
+        {
+            Vector3 planePosition = scrollView.ClippingObject.transform.position + scrollView.ClippingObject.transform.TransformDirection(planeLocalPosition);
 
             Vector3[] points = new Vector3[4];
-            points[0] = center;
-            points[1] = center + rightDelta;
-            points[2] = center + rightDelta - upDelta;
-            points[3] = center - upDelta;
-
-            UnityEditor.Handles.DrawSolidRectangleWithOutline(points, new Color(0.85f, 1.0f, 1.0f, 0.1f), arrowColor);
-            GUIStyle labelStyle = new GUIStyle();
-            labelStyle.normal.textColor = Color.white;
-            UnityEditor.Handles.Label(center + (rightDelta * 0.5f) - (upDelta * 0.5f), new GUIContent("touch plane", "The plane which the finger will need to cross in order for the touch to be calculated as a scroll or release."), labelStyle);
+            points[0] = planePosition - widthHalfExtent + hightHalfExtent;
+            points[1] = planePosition + widthHalfExtent + hightHalfExtent;
+            points[2] = planePosition + widthHalfExtent - hightHalfExtent;
+            points[3] = planePosition - widthHalfExtent - hightHalfExtent;
+
+            float handleSize = HandleUtility.GetHandleSize(planePosition) * 0.30f;
+
+            // Draw handle and plane
+            Handles.color = color;
+            Handles.ArrowHandleCap(0, planePosition, handleRotation, handleSize, EventType.Repaint);
+            Handles.DrawSolidRectangleWithOutline(points, Color.Lerp(color, Color.clear, 0.5f), color);
+
+            // Draw label if handle has smallest vieport distance from the mouse pointer
+            Vector3 handleViewportPosition = SceneView.currentDrawingSceneView.camera.WorldToScreenPoint(planePosition);
+            handleViewportPosition = SceneView.currentDrawingSceneView.camera.ScreenToViewportPoint(handleViewportPosition);
+
+            float mouseHandleDistance = Vector2.Distance(mouseScreenPosition, handleViewportPosition);
+
+            if (mouseHandleDistance < handleSize * 10.0f && mouseHandleDistance < smallestMouseHandleDistance)
```

Comment:
why are we checking for smaller than handleSize * 10?

Reply:
as the handlesize changes scale with scene camera view we want to activate label if mouse is inside a circle with ratio 10 * the current unity handle size